### ISSUE TITLE
Extend DuckDb SQL with timetravel syntax

### ISSUE THREAD

**Issue by Ture2019** [duckdb/duckdb_delta#109]
A major point of delta tables is to enable timetravel queries. An example:
```sql
SELECT *
FROM delta_scan(...)
AS OF '2023-10-23 00:00:00'
```
This is currently missing.

Important reference:
https://docs.databricks.com/en/delta/history.html#delta-time-travel-syntax

**Comment by samansmink** [comment]
good point! this is definitely on our todo list. Right now the delta kernel does not yet support this unfortunately.
